teal/packer/cli.tl

```diff
@@ -0,0 +1,47 @@
+
+local M = {}
+
+local function command_complete(): {string}
+  local actions = require('packer.actions') as {string:function}
+  return vim.tbl_keys(actions)
+end
+
+-- Completion user plugins
+-- Intended to provide completion for PackerUpdate/Sync/Install command
+local function plugin_complete(lead: string, _: string): {string}
+  local plugins = require 'packer.plugin'.plugins
+  local completion_list = vim.tbl_filter(function(name: string): boolean
+    return vim.startswith(name, lead)
+  end, vim.tbl_keys(plugins))
+  table.sort(completion_list)
+  return completion_list
+end
+
+function M.complete(arglead: string, line: string): {string}
+  local words = vim.split(line, '%s+')
+  local n: integer = #words
+
+  local matches: {string} = {}
+  if n == 2 then
+    matches = command_complete()
+  elseif n > 2 then
```

Comment:
I think it might be better to default to command completion, then check the prefix against specific commands for which completing plugin names makes sense? Could do this by giving each command a completion function to call, but that feels like overkill.

Reply:
This is still WIP. I've only asked you to review to check you're happy with going from `PackerInstall` -> `Packer install`. I haven't begun working on the argument design to these actions yet so I've left that blank for now.

Rough sketch in my head: the API for actions will look something like:

```lua
M.install = function(a, b, c, opts, callback)
```

Where:
- `a`, `b`, `c`, (etc) represent positional arguments. Though I don't think we'll have many of these. 
- `opts`  contains any named argument. E.g. `--preview` would be `opts.preview: boolean`.
- `callback` is so users can script packer commands together (as most/all actions are async). This is instead of using the current `PackerComplete` autocmd.

> Could do this by giving each command a completion function to call, but that feels like overkill.

That's exactly the plan. A lot of actions will share completion functions.